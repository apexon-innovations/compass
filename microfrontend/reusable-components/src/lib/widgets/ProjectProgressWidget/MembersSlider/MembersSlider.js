import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Slider from 'react-slick'
import { USER_DUMMY } from '../../../const/imgConst.js'
import style from './MembersSlider.module.scss'
import 'slick-carousel/slick/slick.css'

class MembersSlider extends Component {
  renderSlide(data) {
    return (
      data &&
      data.map((item, key) => {
        return (
          <div className={style.sliderDiv} key={key}>
            <div className={style.member}>
              <img src={item.dp ? item.dp : USER_DUMMY} title={item.name} alt={item.email} />
            </div>
          </div>
        )
      })
    )
  }

  render() {
    const { data } = this.props
    const settings = {
      infinite: data && data.length > 4 ? true : false,
      speed: 800,
      focusOnSelect: true,
      dots: false,
      slidesToShow: 4,
      slidesToScroll: 1,
    }

    return (
      <div className={style.membersSlider}>
        <div className={style.sliderWrap}>
          <Slider {...settings}>{this.renderSlide(data)}</Slider>
        </div>
      </div>
    )
  }
}

MembersSlider.propTypes = {
  data: PropTypes.any,
}

export default MembersSlider
