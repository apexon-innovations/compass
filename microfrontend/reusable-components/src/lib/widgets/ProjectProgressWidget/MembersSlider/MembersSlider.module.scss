@import '../../../theme/theme.scss';

.membersSlider {
  width: 100%;
  position: relative;

  :global(.slick-track) {
    height: auto !important;
  }

  :global(.slick-slide) {
    width: auto;
  }
  :global(.slick-list) {
    width: calc(100% - 50px);
    left: 25px;
  }
  :global(.slick-prev),
  :global(.slick-next) {
    font-size: 0;
    line-height: 0;
    position: absolute;
    top: 50%;
    display: block;
    width: 20px;
    height: 20px;
    padding: 0;
    transform: translate(0, -50%);
    cursor: pointer;
    color: transparent;
    border: none;
    outline: none;
    background: transparent;

    &:before {
      content: '';
      width: 20px;
      height: 20px;
      @include position(absolute, 0, unset, unset, 0);
    }
  }
  :global(.slick-prev) {
    left: 0;
    &:before {
      background: url(../../../images/ic_arrow_left.svg) no-repeat 50% 50%;
    }
  }
  :global(.slick-next) {
    right: 0;
    &:before {
      background: url(../../../images/ic_arrow_right.svg) no-repeat 50% 50%;
    }
  }
}

.sliderWrap {
  max-width: 650px;
  position: relative;
  z-index: 10;
}

.sliderDiv {
  margin: 0 auto;
  max-width: 45px;
}

.member {
  width: 45px;
  height: 45px;
  overflow: hidden;
  @include border-radius(50%);

  img {
    max-width: 100%;
    height: auto;
  }
}
