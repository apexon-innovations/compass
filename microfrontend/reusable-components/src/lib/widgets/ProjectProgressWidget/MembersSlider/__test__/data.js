const data = {
  data: [
    {
      email: 'Pradeep.shrivastava@infostretch.com',
      name: 'Pradeep Shrivastava',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
    },
    {
      email: 'vijay.solanki@infostretch.com',
      name: 'Vijay Solanki',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
    },
    {
      email: 'suraj.bhosale@infostretch.com',
      name: 'Suraj Bhosale',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
    },
    {
      email: 'yatin.bhavsar@infostretch.com',
      name: 'Yatin Bhavsar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
    },
    {
      email: 'vishal.kumar@infostretch.com',
      name: 'Vishal kumar',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
    },
    {
      email: 'gunjan.shah@infostretch.com',
      name: 'Gunjan Shah',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
    },
    {
      email: 'namra.porecha@infostretch.com',
      name: 'Namra Porecha',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
    },
    {
      email: 'purushottam.Khandebharad@infostretch.com',
      name: 'Purushottam Khandebharad',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
    },
    {
      email: 'amit.ghadge@infostretch.com',
      name: 'Amit Ghadge',
      dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
    },
  ],
}

export default data
