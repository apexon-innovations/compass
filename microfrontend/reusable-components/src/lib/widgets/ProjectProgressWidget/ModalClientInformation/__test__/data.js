const data = {
  data: {
    id: '5ee31002d48d514e801cb10b',
    name: 'Compass',
    clientName: 'Infostretch',
    initials: 'C',
    category: 'All (Dev & QA & Support)',
    iconLocation:
      'http://sampleicon.com/image.png',
    billingType: 'Internal',
    startDate: 'Wed Jan 01 00:00:00 UTC 2020',
    endDate: 'Thu Jan 02 00:00:00 UTC 2020',
    resources: {
      actual: 10,
      sow: 10,
    },
    healthMetrics: {
      overall: 'R',
      periodicReview: 'G',
      scope: 'A',
      deliverables: 'A',
      customer: 'G',
      schedule: 'R',
      financials: 'G',
      people: 'G',
      compliance: 'G',
      startUp: 'G',
      process: 'G',
      planning: 'G',
    },
    projectManager: [],
    deliveryManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    accountManager: [
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
        nestProjects: ['Infostretch - DES'],
      },
    ],
    teamMembers: [
      {
        email: 'Pradeep.shrivastava@infostretch.com',
        name: 'Pradeep Shrivastava',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933',
      },
      {
        email: 'vijay.solanki@infostretch.com',
        name: 'Vijay Solanki',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621',
      },
      {
        email: 'suraj.bhosale@infostretch.com',
        name: 'Suraj Bhosale',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950',
      },
      {
        email: 'yatin.bhavsar@infostretch.com',
        name: 'Yatin Bhavsar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973',
      },
      {
        email: 'vishal.kumar@infostretch.com',
        name: 'Vishal kumar',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930',
      },
      {
        email: 'gunjan.shah@infostretch.com',
        name: 'Gunjan Shah',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182',
      },
      {
        email: 'namra.porecha@infostretch.com',
        name: 'Namra Porecha',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375',
      },
      {
        email: 'purushottam.Khandebharad@infostretch.com',
        name: 'Purushottam Khandebharad',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437',
      },
      {
        email: 'amit.ghadge@infostretch.com',
        name: 'Amit Ghadge',
        dp: 'https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269',
      },
    ],
  },
  fileUploaderUrl:
    'https://compass-dev-api.infostretch.com/psr-service/project/5ee31002d48d514e801cb10b/logo',
}

export default data
