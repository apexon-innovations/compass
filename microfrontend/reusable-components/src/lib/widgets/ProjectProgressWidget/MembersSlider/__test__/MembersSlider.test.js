/* eslint-disable */
import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { sleep } from '../../../../../__tests__/commonTestFunction'
import MembersSlider from '../MembersSlider'
import data from './data'
const props = { ...data }

describe(`MembersSlider`, () => {
  let container = null

  beforeEach(() => {
    // set up a DOM element as a render target
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container)
    container.remove()
    container = null
  })

  it(`Snapshot`, async () => {
    act(() => {
      render(<MembersSlider {...props} />, container)
    })
    await act(() => sleep(500))
    expect(container).toMatchSnapshot()
  })
})
