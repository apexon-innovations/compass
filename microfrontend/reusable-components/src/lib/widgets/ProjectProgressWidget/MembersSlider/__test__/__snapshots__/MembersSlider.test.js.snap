// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersSlider Snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="slick-slider slick-initialized"
        dir="ltr"
      >
        <button
          class="slick-arrow slick-prev"
          data-role="none"
          style="display: block;"
          type="button"
        >
           
          Previous
        </button>
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-4"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="gunjan.shah@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                      title="Gunjan Shah"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-3"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="namra.porecha@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                      title="Namra Porecha"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="purushottam.Khandebharad@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                      title="Purushottam Khandebharad"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="amit.ghadge@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                      title="Amit Ghadge"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="Pradeep.shrivastava@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933"
                      title="Pradeep Shrivastava"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="vijay.solanki@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621"
                      title="Vijay Solanki"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="suraj.bhosale@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950"
                      title="Suraj Bhosale"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="3"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="yatin.bhavsar@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973"
                      title="Yatin Bhavsar"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="4"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="vishal.kumar@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930"
                      title="Vishal kumar"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="5"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="gunjan.shah@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                      title="Gunjan Shah"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="6"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="namra.porecha@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                      title="Namra Porecha"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="7"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="purushottam.Khandebharad@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                      title="Purushottam Khandebharad"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="8"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="amit.ghadge@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                      title="Amit Ghadge"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="9"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="Pradeep.shrivastava@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/2755.jpg?77098933"
                      title="Pradeep Shrivastava"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="10"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="vijay.solanki@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/1546.jpg?9447621"
                      title="Vijay Solanki"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="11"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="suraj.bhosale@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3536.jpg?89267950"
                      title="Suraj Bhosale"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="12"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="yatin.bhavsar@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3104.jpg?9939973"
                      title="Yatin Bhavsar"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="13"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="vishal.kumar@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3057.jpg?72778930"
                      title="Vishal kumar"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="14"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="gunjan.shah@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/2136.jpg?32186182"
                      title="Gunjan Shah"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="15"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="namra.porecha@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/545.jpg?23733375"
                      title="Namra Porecha"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="16"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="purushottam.Khandebharad@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3740.jpg?8948437"
                      title="Purushottam Khandebharad"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="17"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="amit.ghadge@infostretch.com"
                      src="https://nest.infostretch.com/nativefw/images/profile_pic/3855.JPG?34957269"
                      title="Amit Ghadge"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="slick-arrow slick-next"
          data-role="none"
          style="display: block;"
          type="button"
        >
           
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
